sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -0,0 +1,441 @@
+                    if not is_datetime(data[column_name]):
+                        datetime_format = column_meta.get(
+                            'datetime_format', column_meta.get('format')
+                        )
+                        data[column_name] = pd.to_datetime(
+                            data[column_name], format=datetime_format
+                        )
+
+                    data.loc[~pd.isna(data[column_name]), column_name] = pd.to_numeric(
+                        data[column_name]
+                    )
+                    data[column_name] = data[column_name].fillna(np.nan)
+
+                continue
+            except Exception as e:
+                message = f'Error: {type(e).__name__} {e}'
+                self._columns_datetime_conversion_failed[column_name] = message
+                LOGGER.debug(message)
+                continue
+
+        return data
+
+    def _discretize_column(self, column_name, data, bin_edges=None):
+        """Discretize a column.
+
+        Args:
+            column_name (str):
+                The name of the column to discretize.
+            data (pandas.Series):
+                The column to discretize.
+            bin_edges (list):
+                The bin edges to use for discretization.
+                Defaults to None.
+
+        Returns:
+            pandas.Series:
+                The discretized column.
+        """
+        column_result = data.copy()
+        try:
+            if bin_edges is None:
+                bin_edges = np.histogram_bin_edges(column_result.dropna())
+
+            column_result = np.digitize(column_result, bins=bin_edges)
+
+        except Exception as e:
+            message = f'Error: {type(e).__name__} {e}'
+            self._columns_discretization_failed[column_name] = message
+            LOGGER.debug(message)
+
+        return column_result, bin_edges
+
+    def _get_processed_data(self, data, metadata):
+        """Get the processed data for the property.
+
+        Preprocess the data by converting datetime columns to numeric and
+        adding a discrete version of datetime and numerical columns to the dataframes.
+
+        Args:
+            data (pandas.DataFrame):
+                The data
+            metadata (dict):
+                The metadata of the table
+        """
+        processed_data = data.copy()
+
+        processed_data = self._convert_datetime_columns_to_numeric(
+            processed_data, metadata
+        )
+
+        for column_name in metadata['columns']:
+            column_meta = metadata['columns'][column_name]
+            column_sdtype = column_meta['sdtype']
+            if column_sdtype in ['numerical', 'datetime']:
+                name_discrete = create_unique_name(column_name + '_discrete', metadata['columns'])
+                processed_data[name_discrete], bin_edges = self._discretize_column(
+                    column_name, processed_data[column_name]
+                )
```

Comment:
I think it might be simpler if we returned the discretized data in its own dataframe and maybe with its own function. This way we wouldn't need the `create_unique_name` method at all and we could just use the column names to easily grab the data from the desired dataframe

Reply:
Yes I was trying to avoid this for memory saving (not saving twice all the category/boolean columns) but if it's not a concern here, I can return the table (original and discretized one).

Reply:
Why would we need to store all the categorical and boolean columns twice? Can't we just take the columns that you're appending to the table and put only those in a separate dataframe?